### PR TITLE
client: ignore "user-agent" if set in outgoing metadata

### PR DESCRIPTION
**Pull request by MakMukhi**

### CODE REVIEW

File: transport/http_util.go [hunk 1]
```diff
@@ -148,7 +149,7 @@ func isReservedHeader(hdr string) bool {
 // that should be propagated into metadata visible to users.
 func isWhitelistedPseudoHeader(hdr string) bool {
 	switch hdr {
-	case ":authority":
+	case ":authority", "user-agent":
```
Comment (ejona86) [inline transport/http_util.go]: Why this change? "user-agent" isn't a pseudo header.

Comment (MakMukhi) [reply transport/http_util.go]: To avoid [this](https://github.com/grpc/grpc-go/blob/7263089022e5a1615620b463eb70f4757697a873/transport/http_util.go#L342) condition so that `user-agent` can be added to metadata exposed to service handler.
Alternatively, we can add a case specifically for `user-agent` in the enclosing `select`. 
Or maybe even rename this function to be `isWhitelisted`instead.
Thoughts?

Comment (ejona86) [reply transport/http_util.go]: Renaming the function to `isWhitelistedHeader` or similar seems good. Tweak the documentation for the function as well.
